### PR TITLE
Harden Story Lab response follow-ups

### PR DESCRIPTION
**Pull request by Phazzie**
## Summary
- harden Story Lab response-status mapping for malformed payloads
- return clean 500 API envelopes when genesis/continuation route engines throw unexpectedly
- tolerate malformed provider error envelopes without throwing or leaking details
- preserve intentional empty raw chapter content and raise Story Lab no-go prompt preservation to the existing 320-character context cap

## Review-thread context
Addresses PR #166 active review threads:
- `PRRT_kwDOPyrwIM6LHlVm` route status malformed payloads
- `PRRT_kwDOPyrwIM6LHlVp` genesis route unexpected throws
- `PRRT_kwDOPyrwIM6LHlVq` continuation route unexpected throws
- `PRRT_kwDOPyrwIM6LHlVr` malformed Story Lab provider errors
- `PRRT_kwDOPyrwIM6LHlVs` empty `rawContent` preservation
- `PRRT_kwDOPyrwIM6LHmfJ` no-go prompt content preservation

## Validation
- RED: `npm run test:story-lab-route-status` failed on null response payload crash
- RED: `npm run test:story-lab-real-engine` failed on empty `rawContent` preservation
- RED: `npm run test:story-service-prompt-guards` failed on no-go content cap
- GREEN: `npm run test:story-lab-route-status`
- GREEN: `npm run test:story-lab-real-engine`
- GREEN: `npm run test:story-service-prompt-guards`
- `npm run test:all`
- `npm run build`
- `git diff --check`
- `scripts/recovery/check-vercel-function-count.sh` (`11/12`)

## Summary by Sourcery

Harden Story Lab API error handling and context limits to improve robustness and safety.

Bug Fixes:
- Ensure Story Lab genesis and continuation routes return clean 500 error envelopes when the underlying engines throw unexpectedly.
- Preserve intentionally empty raw chapter content instead of replacing it with HTML content.
- Cap Story Lab no-go prompt content using the same 320-character context limit as other context fields.
- Handle malformed provider error envelopes by falling back to safe error codes and messages.
- Map malformed or null Story Lab responses to HTTP 500 status codes instead of crashing or misclassifying them.

Enhancements:
- Refactor Story Lab genesis and continuation handlers to be factory-based for easier dependency injection and testing.

Tests:
- Extend Story Lab route status tests to cover malformed payloads and unexpected engine failures.
- Add real-engine tests for empty raw chapter content preservation and safe handling of malformed provider errors.
- Tighten story service prompt guard tests to assert updated no-go content caps.

### CODE REVIEW

File: api/_lib/story-lab/routeStatus.ts [hunk 1]
```diff
@@ -12,16 +12,27 @@ const SERVICE_UNAVAILABLE_CODES = new Set([
...
-  if (payload.success) {
+export function getStoryLabResponseStatus(payload: ApiResponse<unknown> | unknown): number {
+  if (!payload || typeof payload !== 'object') {
+    return 500;
+  }
+
+  const response = payload as Partial<ApiResponse<unknown>>;
+  if (response.success === true) {
     return 200;
   }
```
[the remark this answers is not in the document]
Comment (Phazzie) [reply api/_lib/story-lab/routeStatus.ts]: Resolved via PR #177, merged as e7792bf. getStoryLabResponseStatus now treats success envelopes without data as malformed 500s, with route-status regression coverage.
